### PR TITLE
Change multiline span ASCII art visual order

### PR DESCRIPTION
**Pull request by estebank**
Tweak the ASCII art for nested multiline spans so that we minimize line overlaps.

Partially addresses https://github.com/rust-lang/rust/issues/61017.

### CODE REVIEW

File: compiler/rustc_expand/src/tests.rs [hunk 1]
```diff
@@ -996,21 +996,21 @@ error: foo
...
 11 | ||   X2 Y2 Z2
    | ||__________- `Z` is a good letter too
-...   |
-15 |  | 10
-16 |  |   X3 Y3 Z3
-   |  |_______^ `Y` is a good letter
+...  |
+15 | |  10
+16 | |    X3 Y3 Z3
+   | |________^ `Y` is a good letter
```
Comment (estebank) [inline compiler/rustc_expand/src/tests.rs:1013]: This is the main case I wanted to address.

**Comment by matthiaskrgr** [comment]
@bors r-  clippy tests need to be blessed
https://github.com/rust-lang/rust/pull/104972#issuecomment-1328215921

**Pull request merged**